packages/backend/src/open_payments/payment/incoming/model.ts

```diff
@@ -131,8 +130,15 @@ export class IncomingPayment
     this.receivedAmountValue = amount.value
   }
 
-  public get url(): string {
-    return `${this.paymentPointer.url}${IncomingPayment.urlPath}/${this.id}`
+  public async getUrl(fetchedPaymentPointer?: PaymentPointer): Promise<string> {
```

Comment:
Using `fetchedPaymentPointer` in-case you already have the paymentPointer and don't want to grab it again, like in `toOpenPaymentsType`

Reply:
What if callers just assigned `incomingPayment.paymentPointer = fetchedPaymentPointer`, so that `getUrl`+`getPaymentPointer` only dealt with whether `this.paymentPointer` is defined or not?

Reply:
I'm hesitant to do that, since 
a) we either have to change the type definitions of `getUrl/toOpenPaymentsType` to possibly return `undefined`, or throw inside the method if `paymentPointer` isn't defined
b) it's not entirely clear that the caller has to do that in order to make these methods work.

Maybe we can 
1) have `toOpenPaymentsType` take in a `paymentPointer` (this would also make it sync), it would require passing in `ctx.paymentPointer` in most places but I think that's ok
2) A bit more complicated, but we make a separate `<subresource>WithPaymentPointer` class that constructs a new model with the `paymentPointer` defined, although I'm not sure the additional boilerplate is worth it